### PR TITLE
Add Optimistic Locking to ContactDetailsAPI

### PR DESCRIPTION
**Pull request by LBHCallumM**
## Ticket
[https://hackney.atlassian.net/browse/MR-761?atlOrigin=eyJpIjoiZWM0ZGQxNGI5NGZlNGQ1YmFlNWIzZWQ1NmVmNjNlNDUiLCJwIjoiaiJ9](https://hackney.atlassian.net/browse/MR-761?atlOrigin=eyJpIjoiZWM0ZGQxNGI5NGZlNGQ1YmFlNWIzZWQ1NmVmNjNlNDUiLCJwIjoiaiJ9)

## Summary of Changes

Add optimistic locking to contact details API.

### CODE REVIEW

**Review by humulla — COMMENTED** [review]
You missed out on adding the If-Match header to a few places:
[Startup](https://github.com/LBHackney-IT/contact-details-api/blob/master/ContactDetailsApi/Startup.cs#L196)
[Serverless](https://github.com/LBHackney-IT/contact-details-api/blob/master/ContactDetailsApi/serverless.yml#L40)

**Comment by LBHCallumM** [comment]
> You missed out on adding the If-Match header to a few places: [Startup](https://github.com/LBHackney-IT/contact-details-api/blob/master/ContactDetailsApi/Startup.cs#L196) [Serverless](https://github.com/LBHackney-IT/contact-details-api/blob/master/ContactDetailsApi/serverless.yml#L40)

Good spot

File: ContactDetailsApi.Tests/V1/E2ETests/Fixtures/ContactDetailsFixture.cs [hunk 1]
```diff
@@ -54,12 +58,13 @@ protected virtual void Dispose(bool disposing)
...
             return _fixture.Build<ContactDetailsEntity>()
-                .With(x => x.CreatedBy, () => _fixture.Build<CreatedBy>()
-                    .Create())
+                .With(x => x.CreatedBy, () => _fixture.Build<CreatedBy>().Create())
                 .With(x => x.RecordValidUntil, DateTime.UtcNow)
                 .With(x => x.IsActive, isActive)
                 .With(x => x.TargetType, TargetType.person)
-                .With(x => x.TargetId, TargetId).CreateMany(count);
+                .With(x => x.TargetId, TargetId)
+                .Without(x => x.VersionNumber)
```
Comment (humulla) [inline ContactDetailsApi.Tests/V1/E2ETests/Fixtures/ContactDetailsFixture.cs:66]: I've previously added version number like this ` .With(x => x.VersionNumber, (int?) null)`, not a major different but yeah

**Review by humulla — APPROVED** [review]

File: ContactDetailsApi.Tests/V1/E2ETests/Fixtures/ContactDetailsFixture.cs (hunk 1 above)
Comment (LBHCallumM) [reply ContactDetailsApi.Tests/V1/E2ETests/Fixtures/ContactDetailsFixture.cs:66]: Why do you think that is better?

Comment (Asura5) [reply ContactDetailsApi.Tests/V1/E2ETests/Fixtures/ContactDetailsFixture.cs:66]: I think Callum's version is a lot clearer. explicit cast to nullable int...? hmmm.....

**Pull request merged**